gamemodes/terrortown/gamemode/server/sv_player.lua

```diff
@@ -379,7 +379,7 @@ function GM:KeyPress(ply, key)
 
 		local target = ply:GetObserverTarget()
 
-		if IsValid(target) and target:IsPlayer() then
+		if IsValid(target) and target:IsPlayer() and ply:GetObserverMode() ~= OBS_MODE_ROAMING then -- Only set the spectator's position to the player they are spectating if they are in chase or eye mode. They can use the reload key if they want to return to the person they're spectating
```

Comment:
To be honest, I'd move the comment to a different position and break it into a few lines